b2sdk/transfer/inbound/downloader/parallel.py

```diff
@@ -66,7 +66,14 @@ def is_suitable(self, download_version: DownloadVersion, allow_seeking: bool):
         ) >= 2 and download_version.content_length >= 2 * self.min_part_size
 
     def _get_number_of_streams(self, content_length):
-        return min(self.max_streams, content_length // self.min_part_size) or 1
+        num_streams = content_length // self.min_part_size
+        if self.max_streams is not None:
+            num_streams = min(num_streams, self.max_streams)
+        else:
+            max_threadpool_workers = getattr(self._thread_pool, '_max_workers', None)
+            if max_threadpool_workers is not None:
+                num_streams = min(num_streams, max_threadpool_workers)
```

Comment:
if my understanding is correct, limiting the number of workers to 1 and uploading a file larger than 5GB will cause a crash

Reply:
this is downloading :)

Reply:
ohhhhhh 🤦